theseus/embodied/robot/link.py

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import abc
+from typing import List, Optional, Any
+import torch
+
+
+class Link(abc.ABC):
+    def __init__(
+        self,
+        name: str,
+        id: int = -1,
+        parent: Any = None,
+        children: Optional[List[Any]] = None,
+        dtype: Optional[torch.dtype] = None,
+    ):
+        self._name = name
+        self._id = id
+        self._parent = parent
+        self._children = children if children else []
+        self._ancestors: List[Link] = []
```

Comment:
SInce this is not passed in `__init__` I am guessing this is compute internally. If so should `self.update_ancestors()` be called here?